src/client.rs

```diff
@@ -172,7 +176,23 @@ impl Client {
             return Ok(Response::new(request.uri, Status::Excluded, request.source));
         }
         let status = match request.uri {
-            Uri::Website(ref url) => self.check_website(&url).await,
+            Uri::Website(ref url) => match self.check_website(url).await {
+                Status::Ok(code) if self.prefer_https && url.scheme() == "http" => {
+                    let mut https_url = url.clone();
```

Comment:
Can we move that block into a function; e.g. `is_https_available(&url)`?

Reply:
which module?

Reply:
Actually, this is not exactly `is_https_available`.
The logic is, if `https://*` works, certainly it's available. But sometimes for example it times out due to slow connection, while the website actually provide `https` connection.

So this check is very conservative, it only returns true when `https` connection works at that particular time.

Well, in future we may do some re-tries (which should be configurable by user), for certain cases like time out.

Reply:
Oh, I was just thinking of this section:

```rust
 let mut https_url = url.clone();
  https_url
      .set_scheme("https")
      .unwrap_or_else(|_| unsafe { std::hint::unreachable_unchecked() });
  if self.check_website(&https_url).await.is_success() {
      Status::Error(
          format!(
              "An alternative secure URL is available. Consider replacing {} with {}.",
              &url, &https_url
          ),
          None,
      )
  } else {
      Status::Ok(code)
  }
```

What do you think about think about moving that into a function?

```rust
async fn is_https_available(url: &URL) -> bool {
   // ...
}
```

Reply:
Yes, the code is quite dense -- a function would benefit the code here :+1: